main_test.go

```diff
@@ -191,7 +200,7 @@ func runTest(path, target string, t *testing.T) {
 			cmd = exec.Command(binary)
 		} else {
 			args := append(spec.Emulator[1:], binary)
-			cmd = exec.Command(spec.Emulator[0], args...)
+			cmd = exec.Command(spec.Emulator[0], append(args, additionalArgs...)...)
 		}
 	}
```

Comment:
Here you could do something like:

```go
if len(spec.Emulator) != 0 && spec.Emulator[0] == "wasmtime" {
    for _, v := range environmentVars {
        cmd.Args = append(cmd.Args, "--env", v)
    }
} else {
    cmd.Env = append(cmd.Env, environmentVars...)
}
```

This only sets the environment variables for the newly created process.